### PR TITLE
fix: fixed warehouse error and added validation fir batch quantity

### PR DESCRIPTION
**Pull request by MyuddinKhatri**
Ref: [Asana](https://app.asana.com/0/1192407896547052/1200426337727452)

### CODE REVIEW

File: erpnext/stock/doctype/product_recall/product_recall.py [hunk 1]
```diff
@@ -9,9 +9,17 @@
...
+		self.validate_batch_qty()
+
 	def on_submit(self):
 		self.create_product_recall_notice()
 
+	def validate_batch_qty(self):
+		if not frappe.db.get_value("Batch", self.batch_no, "batch_qty"):
+			frappe.throw(_("Select Batch with Quantity greater than zero."))
+	
+
```
Comment (hrwX) [inline erpnext/stock/doctype/product_recall/product_recall.py]:
```suggestion

```

**Review by hrwX — DISMISSED** [review]

File: erpnext/stock/doctype/product_recall/product_recall.py [hunk 2]
```diff
@@ -9,9 +9,15 @@
...
 class ProductRecall(Document):
+	def validate(self):
+		self.validate_batch_qty()
+
 	def on_submit(self):
 		self.create_product_recall_notice()
 
+	def validate_batch_qty(self):
+		if not frappe.db.get_value("Batch", self.batch_no, "batch_qty"):
+			frappe.throw(_("Select Batch with Quantity greater than zero."))
```
Comment (sahil28297) [inline erpnext/stock/doctype/product_recall/product_recall.py:20]:
```suggestion
			frappe.throw(_("Select Batch with Quantity greater than zero."))

```

File: erpnext/stock/doctype/product_recall/product_recall.py [hunk 3]
```diff
@@ -9,9 +9,15 @@
...
 
 class ProductRecall(Document):
+	def validate(self):
+		self.validate_batch_qty()
+
 	def on_submit(self):
 		self.create_product_recall_notice()
 
+	def validate_batch_qty(self):
+		if not frappe.db.get_value("Batch", self.batch_no, "batch_qty"):
```
Comment (sahil28297) [inline erpnext/stock/doctype/product_recall/product_recall.py:19]: if batch_qty is None, then too it will throw this error, is this intentional?

**Review by sahil28297 — CHANGES_REQUESTED** [review]

File: erpnext/stock/doctype/product_recall/product_recall.py (hunk 3 above)
Comment (MyuddinKhatri) [reply erpnext/stock/doctype/product_recall/product_recall.py:19]: Yes if its None, then it should not proceed.

**Review by sahil28297 — DISMISSED** [review]

**Comment by sahil28297** [comment]
@MyuddinKhatri rebase it to bs-jul-sprint-2 branch

**Review by hrwX — APPROVED** [review]

**Review by sahil28297 — APPROVED** [review]

**Pull request merged**